metabase/metabase
# Convert notebook step components to TypeScript

Epic #28034

We're about to port notebook steps to metabase-lib v2 as a part of #28034. This PR converts most of the notebook step components to TypeScript, so it can assist during refactoring. The work is based on changes from #29554 and #29555

It could be easier to review the PR by commit because converting some of the step components required patching up other code:

* [convert `FilterStep`](https://github.com/metabase/metabase/commit/cf879907c97fb453176d167cf459a042272b5c5d)
* [convert `AggregateStep`](https://github.com/metabase/metabase/commit/27c933df46c763122a4c8cd9b37ee78267299556)
* [convert `BreakoutStep`](https://github.com/metabase/metabase/commit/1b94780f42b273235374e462f7d38f5ae89007f6)
* [convert `SortStep`](https://github.com/metabase/metabase/commit/7d75689a15387495d873b826bd3acb673b0a69c7)
* [convert `DataStep`](https://github.com/metabase/metabase/commit/5dfd0b47bd2adf92e5f00528a29dd5fe5f9612e3)
* [convert `SummarizeStep`](https://github.com/metabase/metabase/commit/b50f4e9a340eab9eab2d609ff3a39ab8f4f2688f)

There are two components that are not yet converted:

* `JoinStep` — seems to be much more complicated, I'd prefer to tackle it once we're porting joins to MLv2
* `LimitStep` — pretty straightforward, but I already have another PR converting it to TS #29618

### How to verify

1. Open `/question/:id/notebook`
2. Try to build some queries, make sure everything works fine (as on `master`)

### Checklist

Existing test suite should keep the refactoring safe


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29644)
